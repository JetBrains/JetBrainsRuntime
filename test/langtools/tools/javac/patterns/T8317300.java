/*
 * @test /nodynamiccopyright/
 * @bug 8317300
 * @summary javac erroneously allows "final" in front of a record pattern
 * @compile/fail/ref=T8317300.out -XDrawDiagnostics T8317300.java
 */
public class T8317300 {
    record Foo (int x) {}
    record Bar (Foo x) {}

    void test1(Object obj) {
        switch (obj) {
            case final Foo(int x) -> {}
            default -> {}
        }
    }

    void test2(Object obj) {
        switch (obj) {
            case Bar(final Foo(int x)) -> {}
            default -> {}
        }
    }
}
