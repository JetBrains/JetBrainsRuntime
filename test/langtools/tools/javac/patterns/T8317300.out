T8317300.java:13:18: compiler.err.mod.not.allowed.here: final
T8317300.java:20:22: compiler.err.mod.not.allowed.here: final
2 errors
